### PR TITLE
libkb: better error reporting for a failed link unpack

### PR DESCRIPTION
**Pull request by maxtaco**
- this code path eventually errors out on hash chain or tail validations, but better to error out earlier
- we were shadowing the error value previously for chain link unpacking
- also clean the code up a bit while we're here

### CODE REVIEW

File: go/libkb/sig_chain.go [hunk 1]
```diff
@@ -329,16 +340,23 @@ func (sc *SigChain) LoadServerBody(m MetaContext, body []byte, low keybase1.Seqn
...
 				return
 			}
 		}
 
 		tail = link
 		numEntries++
 	}, "sigs")
+
 	if err != nil {
+		m.Debug("SigChain#LoadServerBody: failing due to bad chainlink: %s", err)
```
Comment (patrickxb) [inline go/libkb/sig_chain.go]: how do you get here?

Comment (maxtaco) [reply go/libkb/sig_chain.go]: if we set an error at line 314

This isn't very intuitive, but there is no way for the cb from the `ArrayEach` to signify error to the caller

Comment (maxtaco) [reply go/libkb/sig_chain.go]: This was the cause of the original bug here btw...

Comment (patrickxb) [reply go/libkb/sig_chain.go]: oh, i see.  yes, this is not clear at all...ugh.

Comment (maxtaco) [reply go/libkb/sig_chain.go]: actually my fix is no good, since a nil error can stomp over a non-nil error

Comment (patrickxb) [reply go/libkb/sig_chain.go]: should bail out if `inErr` isn't nil too...

**Pull request merged**